Do not interpret DeferredBehaviors on termination

2m opened the pull request:
The actual fix is in `SupervisionMechanics.scala`. Other changes are to unify undefer usage across various code places.

Fixes #22687

2m committed:
#22687 Undefer before terminate

patriknw commented on akka-typed/src/main/scala/akka/typed/Behavior.scala:
```diff
@@ -179,12 +175,8 @@ object Behavior {
-      case x                                 ⇒ x
-    }
-  }
+  def preStart[T](behavior: Behavior[T], ctx: ActorContext[T]): Behavior[T] =
```
I find the name `preStart` a bit confusing nowadays. Could you try to remove it and directly call `validateAsInitial` instead, since that is the purpose. Then `validateAsInitial` should probably perform the `undefer` as the first thing.

2m committed:
#22744 Remove preStart

2m commented:
Removed preStart. Instead of adding undefer to validateAsInitial, I have replaced preStart with the composition of validateAsInitial and undefer. The reason was that validateAsInitial is used in some places where context is not accessible, and therefore it is not possible to call undefer from there.
